src/openrct2-ui/windows/Staff.cpp

```diff
@@ -119,1240 +121,1169 @@ static rct_widget* window_staff_page_widgets[] = {
+            if (viewport->flags & VIEWPORT_FLAG_SOUND_ON)
+            {
+                gfx_draw_sprite(dpi, ImageId(SPR_HEARING_VIEWPORT), windowPos + ScreenCoordsXY{ 2, 2 });
+            }
+        }
 
-    auto peep = GetStaff(w);
-    if (peep == nullptr)
-    {
-        return;
+        // Draw the centred label
+        if (_staff == nullptr)
+        {
+            return;
+        }
+        auto ft = Formatter();
+        _staff->FormatActionTo(ft);
+        const auto& widget = widgets[WIDX_BTM_LABEL];
+        auto screenPos = windowPos + ScreenCoordsXY{ widget.midX(), widget.top };
+        int32_t widgetWidth = widget.width();
+        DrawTextEllipsised(dpi, screenPos, widgetWidth, STR_BLACK_STRING, ft, { TextAlignment::CENTRE });
     }
-    if (peep->WindowInvalidateFlags & PEEP_INVALIDATE_STAFF_STATS)
+
+    void DrawOverviewTabImage(rct_drawpixelinfo* dpi)
     {
-        peep->WindowInvalidateFlags &= ~PEEP_INVALIDATE_STAFF_STATS;
-        w->Invalidate();
-    }
-}
+        if (WidgetIsDisabled(this, WIDX_TAB_1))
+            return;
 
-/**
- *
- *  rct2: 0x6BEC80, 0x6BE9DA
- */
-void WindowStaffUnknown05(rct_window* w)
-{
-    widget_invalidate(w, WIDX_TAB_1);
-}
+        const auto& widget = widgets[WIDX_TAB_1];
+        int32_t widgetWidth = widget.width() - 1;
+        int32_t widgetHeight = widget.height() - 1;
+        auto screenCoords = windowPos + ScreenCoordsXY{ widget.left + 1, widget.top + 1 };
+        if (page == WINDOW_STAFF_OVERVIEW)
+            widgetHeight++;
 
-/**
- *
- *  rct2: 0x006BE9E9
- */
-void WindowStaffStatsInvalidate(rct_window* w)
-{
-    ColourSchemeUpdateByClass(w, static_cast<rct_windowclass>(WC_STAFF));
+        rct_drawpixelinfo clip_dpi;
+        if (!clip_drawpixelinfo(&clip_dpi, dpi, screenCoords, widgetWidth, widgetHeight))
+        {
+            return;
+        }
 
-    if (window_staff_page_widgets[w->page] != w->widgets)
-    {
-        w->widgets = window_staff_page_widgets[w->page];
-        WindowInitScrollWidgets(w);
-    }
+        screenCoords = ScreenCoordsXY{ 14, 20 };
 
-    w->pressed_widgets |= 1ULL << (w->page + WIDX_TAB_1);
+        if (_staff == nullptr)
+        {
+            return;
+        }
 
-    const auto peep = GetStaff(w);
-    if (peep == nullptr)
-    {
-        return;
-    }
+        if (_staff->Is<Staff>() && _staff->AssignedStaffType == StaffType::Entertainer)
+            screenCoords.y++;
 
-    auto ft = Formatter::Common();
-    peep->FormatNameTo(ft);
+        int32_t ebx = GetPeepAnimation(_staff->SpriteType).base_image + 1;
 
-    window_staff_stats_widgets[WIDX_BACKGROUND].right = w->width - 1;
-    window_staff_stats_widgets[WIDX_BACKGROUND].bottom = w->height - 1;
+        int32_t eax = 0;
 
-    window_staff_stats_widgets[WIDX_RESIZE].right = w->width - 1;
-    window_staff_stats_widgets[WIDX_RESIZE].bottom = w->height - 1;
+        if (page == WINDOW_STAFF_OVERVIEW)
+        {
+            eax = var_496;
```

Comment:
Replace all occurrences of `var_496` with a new private field in this class. Rename to something better, like tabAnimationOffset.